sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -20,7 +20,41 @@
 from ._helpers import _host_only, _is_tag, _strip_alg
 
 
-class ArtifactArchitecture(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+class TypeInsensitiveEnumMeta(CaseInsensitiveEnumMeta):
+    """Extensible enum metaclass to trick extended enums into pretending to be the same type."""
+    def __instancecheck__(cls, other):
+        is_instance = super().__instancecheck__(other)
+        if not is_instance:
+            # Let's check if this is an instance of an extended enum. To do this we compare the enum member maps
+            # excluding the current attribute, along with the class name.
+            try:
+                unextended = {k: v for k, v in other._member_map_.items() if v != other.value}
+                return unextended == cls._member_map_ and cls.__name__ == other.__class__.__name__
+            except:  # pylint:disable=bare-except
+                pass
+        return is_instance
+
+
+class EnumBase(str, Enum, metaclass=TypeInsensitiveEnumMeta):
+    """Extensible enum base to allow for unrecognized strings to be returned in the payload without
+    breaking any existing code.
+    """
+    @classmethod
+    def _extended(cls, value):
+        try:
+            return cls(value)
+        except ValueError:
+            if value in [None, ""]:
+                return None
+            value = str(value)
+            name = "".join(c for c in value.upper() if c.isalnum()).rstrip()
+            extended_values = {m.name: m.value for m in cls}
+            extended_values[name] = value
+            ExtendedEnum = EnumBase(cls.__name__, extended_values)
+            return ExtendedEnum(value)
```

Comment:
This will need some more conservative error handling in cases of invalid member names or values.
If it all falls over, will probably just fallback to returning `str(value)` rather than raise to protect deserialization.